common/etc/nginx/include/s3gateway.js

```diff
@@ -994,6 +1003,38 @@ async function _fetchEC2RoleCredentials() {
     };
 }
 
+/**
+ * Get the credentials by assuming calling AssumeRoleWithWebIdentity with the environment variable
+ * values ROLE_ARN, AWS_WEB_IDENTITY_TOKEN_FILE and HOSTNAME
+ *
+ * @returns {Promise<{accessKeyId: (string), secretAccessKey: (string), sessionToken: (string), expiration: (string)}>}
+ * @private
+ */
+async function _fetchWebIdentityCredentials(r) {
+    var arn   = process.env['AWS_ROLE_ARN'];
+    var name  = process.env['HOSTNAME'] || 'nginx-s3-gateway';
+    var token = fs.readFileSync(process.env['AWS_WEB_IDENTITY_TOKEN_FILE']);
+    
+    var params = "Version=2011-06-15&Action=AssumeRoleWithWebIdentity&RoleArn=" + arn + "&RoleSessionName=" + name + "&WebIdentityToken=" + token;
+
+    var response = await ngx.fetch("https://sts.us-east-1.amazonaws.com?" + params, {
```

Comment:
_By default, the AWS Security Token Service (AWS STS) is available as a global service, and all STS requests go to a single endpoint at https://sts.amazonaws.com/. AWS recommends using Regional STS endpoints to reduce latency, build in redundancy, and increase session token validity._

https://docs.aws.amazon.com/general/latest/gr/sts.html#sts_region

I would recommend adding an environment AWS_STS_ENDPOINT to configure the desired endpoint, while defaulting to sts.amazonaws.com.

Reply:
Hello @mvanholsteijn, sure I will make the changes accordingly.

Reply:
I agree with the above assessment.